internal/sdkprovider/service/vpc/azure_org_vpc_peering_connection.go

```diff
@@ -0,0 +1,221 @@
+	return &schema.Resource{
+		Description:   "Creates and manages an Azure VPC peering connection with an Aiven VPC.",
+		CreateContext: common.WithGenClientDiag(resourceAzureOrgVPCPeeringConnectionCreate),
+		ReadContext:   common.WithGenClientDiag(resourceAzureOrgVPCPeeringConnectionRead),
+		DeleteContext: common.WithGenClientDiag(resourceAzureOrgVPCPeeringConnectionDelete),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: schemautil.DefaultResourceTimeouts(),
+
+		Schema: aivenAzureOrgVPCPeeringConnectionSchema,
+	}
+}
+
+func resourceAzureOrgVPCPeeringConnectionCreate(ctx context.Context, d *schema.ResourceData, client avngen.Client) diag.Diagnostics {
+	var (
+		orgID               = d.Get("organization_id").(string)
+		vpcID               = d.Get("organization_vpc_id").(string)
+		azureSubscriptionID = d.Get("azure_subscription_id").(string)
+		vnetName            = d.Get("vnet_name").(string)
+		appID               = d.Get("peer_azure_app_id").(string)
+		tenantID            = d.Get("peer_azure_tenant_id").(string)
+		resourceGroup       = d.Get("peer_resource_group").(string)
+
+		req = organizationvpc.OrganizationVpcPeeringConnectionCreateIn{
+			PeerAzureAppId:    util.ToPtr(appID),
+			PeerAzureTenantId: util.ToPtr(tenantID),
+			PeerCloudAccount:  azureSubscriptionID,
+			PeerResourceGroup: util.ToPtr(resourceGroup),
+			PeerVpc:           vnetName,
+		}
+	)
+
+	pCon, err := createPeeringConnection(ctx, orgID, vpcID, client, d, req)
+	if err != nil {
+		return diag.Errorf("Error creating VPC peering connection: %s", err)
+	}
+
+	diags := getDiagnosticsFromState(newOrganizationVPCPeeringState(pCon))
+
+	d.SetId(schemautil.BuildResourceID(orgID, vpcID, pCon.PeerCloudAccount, pCon.PeerVpc, pCon.PeerResourceGroup))
+
+	// in case of an error delete VPC peering connection
+	if diags.HasError() {
+		deleteDiags := resourceAzureOrgVPCPeeringConnectionDelete(ctx, d, client)
+		d.SetId("") // Clear the ID after delete
+
+		return append(diags, deleteDiags...)
+	}
+
+	return append(diags, resourceAzureOrgVPCPeeringConnectionRead(ctx, d, client)...)
+}
+
+func resourceAzureOrgVPCPeeringConnectionRead(ctx context.Context, d *schema.ResourceData, client avngen.Client) diag.Diagnostics {
+	orgID, vpcID, cloudAccount, vnetName, resourceGroup, err := schemautil.SplitResourceID5(d.Id())
+	if err != nil {
+		return diag.Errorf("error parsing Azure peering VPC ID: %s", err)
+	}
+
+	vpc, err := client.OrganizationVpcGet(ctx, orgID, vpcID)
+	if err != nil {
+		if avngen.IsNotFound(err) {
+			return diag.FromErr(schemautil.ResourceReadHandleNotFound(err, d))
+		}
+
+		return diag.Errorf("failed to get VPC with ID %q: %s", vpcID, err)
+	}
+
+	pc := lookupAzurePeeringConnection(vpc, cloudAccount, vnetName, resourceGroup)
+	if pc == nil {
+		return diag.FromErr(schemautil.ResourceReadHandleNotFound(errors.New("VPC peering connection not found"), d))
```

Comment:
`ResourceReadHandleNotFound` won't recognize this error, and won't clear the id.

Reply:
Fixed